python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -184,13 +189,20 @@ def evaluate(
         df_grouped_stats = pd.DataFrame()
         # defining variables to compute running mean and variance for t-test computations
         agg_count, agg_mean, agg_M2 = 0, 0, 0
+        group_metric_running_variance_params = {}
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
             # Accumulating statistics for t-test calculation using 1/rank
             clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
-            diff = (1/clicked_records[RankingConstants.NEW_RANK] - 1/clicked_records[
-                self.feature_config.get_rank("node_name")]).to_list()
+            clicked_records[RankingConstants.NEW_MRR] = 1.0 / clicked_records["new_rank"]
+            clicked_records[RankingConstants.OLD_MRR] = 1.0 / clicked_records[self.feature_config.get_rank("node_name")]
+            # Statistical analysis pre-processing
+            if len(group_key) > 0 and len(eval_dict["metrics"]) > 0:
+                statistical_analysis_preprocessing(clicked_records, group_metric_running_variance_params, group_key,
+                                                   eval_dict["variance_list"])
+
+            diff = (clicked_records[RankingConstants.NEW_MRR] - clicked_records[RankingConstants.OLD_MRR]).to_list()
```

Comment:
@mohazahran `ranking_model` shouldn't be hardcoding these MRR metrics. Move it out to the `metrics_helper`.